src/Atmos/Model/source.jl

```diff
@@ -9,8 +9,11 @@ end
 function atmos_source!(::Nothing, atmos::AtmosModel, source::Vars, state::Vars, diffusive::Vars, aux::Vars, t::Real)
 end
 # sources are applied additively
-function atmos_source!(stuple::Tuple, atmos::AtmosModel, source::Vars, state::Vars, diffusive::Vars, aux::Vars, t::Real)
-  map(s -> atmos_source!(s, atmos, source, state, diffusive, aux, t), stuple)
```

Comment:
No opinion on what is cleaner, but you could have done:

```
ntuple(Val(length(stuple)) do i
  Base.@_inline_meta # this is just defensive programming on my part
  s = @inbounds stuple[i]
  atmos_source!(s, atmos, source, state, diffusive, aux, t)
  nothing
end
```

Reply:
probably good if we can do without `@generated`.